applets/clock/set-timezone.c

```diff
@@ -202,6 +202,7 @@ set_time_notify (GObject *source_object,
 		if (data->callback)
 			data->callback (data->data, NULL);
 	}
+	free_data (data);
```

Comment:
It's already free on `set_system_time_async` and  `set_system_timezone_async` after calling `set_time_async` in applets/clock/set-timezone.c

Reply:
Nope, there's internal reference counting. This should've been a proper object, it's not obvious on the outside.

Reply:
You are right, I'm sorry, `free_data` checks reference counting before freeing the object ;)